examples/bigbench_hard/bigbench_hard.py

```diff
@@ -0,0 +1,361 @@
+                new_state["phase"] = 2
+                new_states.append(new_state)
+            else:
+                raise ValueError(f"parse_generate_answer: Unknown method: {state['method']}")
+        return new_states
+
+    def parse_aggregation_answer(self, response: str, **kwargs) -> Union[Dict, List[Dict]]:
+        pass
+
+    def parse_improve_answer(self, response: str, **kwargs) -> Dict:
+        pass
+
+    def parse_validation_answer(self, response: str, **kwargs) -> bool:
+        pass
+
+    def parse_score_answer(self, response: str, **kwargs) -> List[float]:
+        pass
+
+
+def io() -> operations.GraphOfOperations:
+    """
+    Generates the Graph of Operations for the IO method.
+
+    :return: Graph of Operations
+    :rtype: GraphOfOperations
+    """
+    operations_graph = operations.GraphOfOperations()
+
+    operations_graph.append_operation(operations.Generate(1, 1))
+    operations_graph.append_operation(operations.GroundTruth(test_answer))
+
+    return operations_graph
+
+
+def cot() -> operations.GraphOfOperations:
+    """
+    Generates the Graph of Operations for the COT method.
+
+    :return: Graph of Operations
+    :rtype: GraphOfOperations
+    """
+    operations_graph = operations.GraphOfOperations()
+
+    operations_graph.append_operation(operations.Generate(1, 1))
+    operations_graph.append_operation(operations.GroundTruth(test_answer))
+
+    return operations_graph
+
+
+def cot_zeroshot() -> operations.GraphOfOperations:
+    """
+    Generates the Graph of Operations for the COT method.
+
+    :return: Graph of Operations
+    :rtype: GraphOfOperations
+    """
+    operations_graph = operations.GraphOfOperations()
+
+    operations_graph.append_operation(operations.Generate(1, 1))
+    operations_graph.append_operation(operations.GroundTruth(test_answer))
+
+    return operations_graph
+
+
+def cot_sc() -> operations.GraphOfOperations:
+    """
+    Generates the Graph of Operations for the COT-SC method.
+
+    :return: Graph of Operations
+    :rtype: GraphOfOperations
+    """
+    num_branches = 5
+    operations_graph = operations.GraphOfOperations()
+
+    generate_operation = operations.Generate(1, num_branches)
+    operations_graph.append_operation(generate_operation)
+    operations_graph.append_operation(
+        operations.Selector(selector=score_answers_by_frequency))  # have to do this less than ideal implementation
+    # due to issues with existing scoring function, might be better to refactor
+    operations_graph.append_operation(operations.KeepBestN(1))
+    operations_graph.append_operation(operations.GroundTruth(test_answer))
+
+    return operations_graph
+
+
+def tot() -> operations.GraphOfOperations:
+    """
+     Generates the Graph of Operations for the TOT method.
+
+     :return: Graph of Operations
+     :rtype: GraphOfOperations
+     """
+    raise NotImplementedError("Not Implemented")
+
+
+def run(
+        data_ids: List[int],
```

Comment:
you should implement using the data_ids somehow or in some other way make sure that you can select the number of samples

Reply:
at the moment your code runs on all examples. This is very costly for testing your code